modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2786,6 +2799,26 @@ Error GDScriptCompiler::_prepare_compilation(GDScript *p_script, const GDScriptP
 				p_script->constants.insert(name, enum_value.value);
 			} break;
 
+			case GDScriptParser::ClassNode::Member::STRUCT: {
+				if (!member.m_struct) {
+					break;
+				}
+				const GDScriptParser::StructNode &struct_node = *member.m_struct;
+				StructInfo struct_info = StructInfo(struct_node.identifier->name, struct_node.members.size());
+				for (int j = 0; j < struct_info.count; j++) {
+					const GDScriptParser::VariableNode *struct_member = struct_node.members[j];
+					Variant default_value = struct_member->initializer ? struct_member->initializer->reduced_value : Variant(); // TODO: need better logic here.
+					struct_info.set(j,
+							struct_member->identifier->name,
+							struct_member->datatype.builtin_type,
+							struct_member->datatype.native_type,
+							nullptr,
+							default_value);
+					//TODO: does not yet support nested structs.
+				}
+				p_script->structs.insert(struct_info.name, struct_info);
+			} break;
+
```

Comment:
@dalexeev @vnen Whenever either of you have time, I'd love some help on how I should be handling struct defaults. The current approach here is to store the reduced value of the initializer expression. The major problem with this approach is that It does not work for reference types like `Dictionary` or `Object` because every default struct will have a reference to the same value of the reference type, instead of a fresh copy. 

I think the solution is to compile something like an `implicit_new` function and then call it to obtain a fresh default struct. I'm having trouble implementing that approach however, because I'm not quite sure how `implicit_new` functions are compiled. It seems like it occurs in `_parse_function`, but that seems to create the constructor for the whole class, whereas I just want the constructor for a struct within the class. Any tips?